jsonrpc/server.go

```diff
@@ -359,20 +377,33 @@ func (s *Server) handleBatchRequest(ctx context.Context, batchReq []json.RawMess
 					resp.ID = req.ID
 				}
 			}
-			// for notification request response is nil
+			// for notification request response is nil and header is irrelevant for now
 			if resp != nil {
-				addResponse(resp)
+				addResponse(resp, header)
 			}
 		})
 	}
 
 	wg.Wait()
+
+	// merge headers
+	finalHeaders := http.Header{}
+	for _, header := range headers {
+		for k, v := range header {
+			for _, e := range v {
+				finalHeaders.Add(k, e)
+			}
+		}
+	}
+
 	// according to the spec if there are no response objects server must not return empty array
 	if len(responses) == 0 {
-		return nil, nil
+		return nil, finalHeaders, nil
 	}
 
-	return json.Marshal(responses)
+	result, err := json.Marshal(responses)
+
+	return result, finalHeaders, err // todo: fix batch request aggregate header
```

Comment:
Does this ToDo need fixed before merging into main?

Reply:
no it doesn't its there to say the way handle batch request is now isn't optimal, better to refactor later.